Добавлена механика появления диалогового окна при приближении к персонажу и само диалоговое окно

SwitcheRka opened the pull request:
Added dialogue system.

SwitcheRka committed:
Добавлена механика появления диалогового окна при приближении к персонажу и само диалоговое окно

Somestrangeer commented:
Диалоговое окно слишком большое, нужно уменьшить, чтобы не выходило за пределы камеры. Нужно добавить появление окна нажатием кнопки E, так же вернуть назад переменные здоровья и урона. 
Создай абстракцию, интерфейс для диалогов.
Как должно работать: если я создаю объекты, которые имеют тэг npc и я, подходя к одному из них, должен увидеть подсказку "нажмите Е для начала диалога", после в коде идет проверка, с каким нпс мы заговорили, а это в свою очередь ведет нас к созданию интерфейса для контроля диалогов, где хранятся текста диалогов.
![Screenshot 2024-02-06 091732](https://github.com/Somestrangeer/GameProject/assets/95109481/1fcf7d71-eae0-41c2-a002-6530c8c94451)

SwitcheRka committed:
Переписан и дополнен код диалоговой системы

Переписал код, добавил функцию старта диалога по кнопке E, перелистывание предложений теперь так же происходит по кнопке (R). Примитивные анимации подсвечивания спрайта когда игрок входит в зону взаимодействия с ним (в зону, в которой можно начать диалог) и анимация появления диалогового окна.

SwitcheRka committed:
Еще раз переделана диалоговая система (не полностью)

-вылез баг, при появлении 2го нпс и т.д. игра ломается и показывает диалоги изначального нпс.
-не все комментарии
Из положительного: Инк файлы работают и инфа из них спокойно переносится в игру (за исключением описанного выше бага). Так же появилась возможность выбора ответа (не доделана, ответы пока просто отображаются)

SwitcheRka committed:
Добавлены ивенты к кнопкам и т.д. (подробнее в описании)

Добавлены ивенты к кнопкам в диалогах (выбираются клавишами вверх-вниз и нажимаются клавишей Enter). Пофикшен баг с появлением не того диалога.  Добавлены комментарии к коду.

SwitcheRka committed:
Final commit

Final commit
